src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -832,11 +851,21 @@ ErrorIfInsertPartitionColumnDoesNotMatchSelect(Query *query, RangeTblEntry *inse
 				break;
 			}
 
+			cteListContext = list_make1(query->cteList);
+			originalRelationId = FindOriginalRelationId(subqeryTargetEntry->expr,
```

Comment:
This code portion checks if the Var corresponding to the partition column on the target table is actually a partition column in the source table.

When we are selecting from a view, TargetEntry->resorigtbl points to a view, and partition method or partition column calls fails due to that. I needed to do a recursive search based on target entry Var->attno to determine which relation it is coming from to perform partition column checks.

CTE involvement is unintended side effect here. If query contains a CTE we need to be able to traverse the CTE to be able to reach the source relation.